File: js/tests/performance.test.js
const { performance } = require('perf_hooks');
const count = require('../countWellFormedParenthesis');

test('Performance test for n = 15', () => {
    const n = 15;
    const start = performance.now();
    const result = count(n);
    const end = performance.now();

    const duration = end - start;

    console.log(`Result: ${result}`);
    console.log(`Execution time: ${duration.toFixed(3)} ms`);

    expect(result).toBeGreaterThan(0);
    expect(duration).toBeLessThan(200); // 200 ms предел
});

File: src/test/java/peaksoft/gts/gts_test/PerformanceTest.java
package peaksoft.gts.gts_test;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PerformanceTest {
    @Test
    void testPerformance() {
        long start = System.nanoTime();
        int result = Count.countWellFormatedParenthesis(15);
        long end = System.nanoTime();

        System.out.println("Result: " + result);
        System.out.println("Execution time (ms): " + (end - start) / 1_000_000.0);

        assertTrue(end - start < 1_000_000_000); // < 1 сек
    }
}